modules/frontend/tag_handlers_test.go

```diff
@@ -167,6 +170,57 @@ func runnerTagValuesV2ClientCancelContext(t *testing.T, f *QueryFrontend) {
 	require.Equal(t, status.Error(codes.Canceled, "context canceled"), err)
 }
 
+func TestSearchTagsV2AlwaysReturnsIntrinsics(t *testing.T) {
```

Comment:
Does this test check that the intrinsics are returned even when we set low MaxBytesPerTagValuesQuery ?

Reply:
+1, I would like to a see a test that makes sure that we return intrinsics even when limit is hit, it would catch this bug if we introduce it again, and imo it would be very easy to introduce this bug again during a frontend refactor.

Reply:
Updated the test to check both within or over the limit.